### PR TITLE
5.1 task solution

### PR DESCRIPTION
**Pull request by gradiviro**

### CODE REVIEW

File: src/main/java/core/basesyntax/Car.java [hunk 1]
```diff
@@ -1,5 +1,23 @@
...
+        this.color = color;
+        this.model = model;
+        this.price = price;
+        this.passengerCapacity = passengerCapacity;
+        this.gasType = gasType;
+    }
+
+    @Override
+    public Car clone() {
+        return new Car(color, model, price, passengerCapacity, gasType);
```
Comment (RomanDubovskyi) [inline src/main/java/core/basesyntax/Car.java:21]: what about Engine?

File: src/main/java/core/basesyntax/Car.java [hunk 2]
```diff
@@ -1,5 +1,23 @@
...
+        this.color = color;
+        this.model = model;
+        this.price = price;
+        this.passengerCapacity = passengerCapacity;
+        this.gasType = gasType;
+    }
+
+    @Override
+    public Car clone() {
+        return new Car(engine,color, model, price, passengerCapacity, gasType);
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/Car.java:21]: pay attention to the engine. Think, what means a deep copy

File: src/main/java/core/basesyntax/Car.java [hunk 3]
```diff
@@ -1,5 +1,24 @@
...
-public class Car {
+public class Car implements Cloneable {
+    private Engine engine;
+    private String color;
+    private String model;
+    private int price;
+    private int passengerCapacity;
+    private String gasType;
 
+    public Car(Engine engine, String color, String model, int price, int passengerCapacity, String gasType) {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:11]: pay attention to this line, probably that the reason of travis failing

File: src/main/java/core/basesyntax/Engine.java [hunk 4]
```diff
@@ -1,5 +1,23 @@
...
+                  String manufacturer, String type) {
+        this.capacity = capacity;
+        this.valveNumber = valveNumber;
+        this.cylinderNumber = cylinderNumber;
+        this.manufacturer = manufacturer;
+        this.type = type;
+    }
+
+    @Override
+    public Engine clone() throws CloneNotSupportedException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Engine.java:20]: does it really throws that exception or it has a grey color?

File: src/main/java/core/basesyntax/Car.java [hunk 5]
```diff
@@ -1,5 +1,25 @@
...
+        this.engine = engine;
+        this.color = color;
+        this.model = model;
+        this.price = price;
+        this.passengerCapacity = passengerCapacity;
+        this.gasType = gasType;
+    }
+
+    @Override
+    public Car clone() throws CloneNotSupportedException {
```
Comment (dmytro-hryhoruk) [inline src/main/java/core/basesyntax/Car.java:22]: same

**Pull request closed without merging**